### PR TITLE
Fix BatchNorm CUDA kernel definition

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: The type constraint in the CUDA BatchNorm kernel definition contains names of the inputs instead of the type template in the spec. Noticed this during docs refresh.

### CODE REVIEW

File: docs/OperatorKernels.md [hunk 1]
```diff
@@ -422,16 +422,8 @@
...
-| | ||**X** = tensor(double), tensor(float), tensor(float16)|
-| | ||**mean** = tensor(double), tensor(float), tensor(float16)|
-| | ||**scale** = tensor(double), tensor(float), tensor(float16)|
-| | ||**var** = tensor(double), tensor(float), tensor(float16)|
-| | |[7, 8]|**B** = tensor(double), tensor(float), tensor(float16)|
-| | ||**X** = tensor(double), tensor(float), tensor(float16)|
-| | ||**mean** = tensor(double), tensor(float), tensor(float16)|
-| | ||**scale** = tensor(double), tensor(float), tensor(float16)|
-| | ||**var** = tensor(double), tensor(float), tensor(float16)|
+|BatchNormalization|(*in* X:**T**, *in* scale:**T**, *in* B:**T**, *in* mean:**T**, *in* var:**T**, *in* training_mode:**T1**, *out* Y:**T**, *out* output_mean:**T**, *out* output_var:**T**, *out* saved_mean:**T**, *out* saved_var:**T**) or (*in* X:**T**, *in* scale:**T**, *in* B:**T**, *in* mean:**T**, *in* var:**T**, *out* Y:**T**, *out* mean:**T**, *out* var:**T**, *out* saved_mean:**T**, *out* saved_var:**T**)|9+|**T** = tensor(double), tensor(float), tensor(float16)|
```
Comment (hariharans29) [inline docs/OperatorKernels.md:425]: Actual doc change. Others are just changes that have come in since last refresh,

File: onnxruntime/core/providers/cuda/nn/batch_norm.cc [hunk 2]
```diff
@@ -19,11 +19,7 @@ namespace cuda {
       T,                                                             \
       kCudaExecutionProvider,                                        \
       KernelDefBuilder()                                             \
-          .TypeConstraint("X", DataTypeImpl::GetTensorType<T>())     \
-          .TypeConstraint("scale", DataTypeImpl::GetTensorType<T>()) \
-          .TypeConstraint("B", DataTypeImpl::GetTensorType<T>())     \
-          .TypeConstraint("mean", DataTypeImpl::GetTensorType<T>())  \
-          .TypeConstraint("var", DataTypeImpl::GetTensorType<T>()),  \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()),    \
```
Comment (hariharans29) [inline onnxruntime/core/providers/cuda/nn/batch_norm.cc:22]: Kernel def change
